PolyEngine/Engine/Src/Engine.cpp

```diff
@@ -11,6 +11,9 @@ Engine::Engine()
 {
 	ASSERTE(gEngine == nullptr, "Creating engine twice?");
 	gEngine = this;
+
+	gAssetsPathConfig.Load();
+	gDebugConfig.Load();
```

Comment:
I'm not sure about having to remember to manually load the configuration, given that you get the defaults if you forget to. Sounds like a future common source of silent config-overwriting bugs.

Reply:
Well, now this is simply part of initialization process. In the future it would be good to implement some config manager for this. The problem is, some configuration files need to be loaded first.

Reply:
Resolving a dependency tree, hmm... Might be not trivial.